### DISCUSSION TITLE
Local extensions directory type restructure

### DISCUSSION THREAD

**Discussion by dimitrov-adrian** [directus/directus#7357] [Feature Requests]
Looking at the remote (npm installable extensions) and local ones. They provide different installation experience, and since `directus:extension` manifest is available through the `package.json`, I think unifying will be better here.

So for example, right now we have

- ./extensions/
  - interfaces/
    - my-interface/
      - index.js   <--- the entry point
  - displays
  ...
  - hooks
    - my-hook-extension/
      - index.js  <--- the entrypoint

My idea is to have

- ./extensions/
  - my-extension-interface
    - index.js
    - package.json 
  - my-hook-extension
    - dist/
      - build.js
    - package.json

Where the corresponding `package.json` will be required and then `directus:extension` manifest will be utilised

I think, the benefits at all will be unified extension development, of course also using same storage during extensions in docker instance.

**Comment by Attacler** [comment]
I think this would be a good approach but instead of using 'Extensions' i would name the directory differend (plugins for example).
If we dont do this the current extensions approach would cause issues.

**Reply by benhaynes** [reply]
@rijkvanzanten — thoughts on this direction?

**Reply by rijkvanzanten** [reply]
We wouldn't want to have two different folders going at the same time that serve the same purpose. To me, the extensions folder as is is mostly for local development and tiny extensions, with the npm installed extensions being the primary target for more in-depth multi-facetted extensions 👍🏻

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
